C/plugins/north/PI_Server_V2/plugin.cpp

```diff
@@ -0,0 +1,269 @@
+{
+	/**
+	 * Handle the PI Server parameters here
+	 */
+	ConfigCategory configCategory("cfg", configData["GLOBAL_CONFIGURATION"]);
+	string url = configCategory.getValue("URL");
+	unsigned int timeout = atoi(configCategory.getValue("OMFHttpTimeout").c_str());
+	string producerToken = configCategory.getValue("producerToken");
+
+	string formatNumber = configCategory.getValue("formatNumber");
+	string formatInteger = configCategory.getValue("formatInteger");
+
+	/**
+	 * Handle the OMF_TYPES parameters here
+	 */
+	ConfigCategory configTypes("types", configData["OMF_TYPES"]);
+	string typesId = configTypes.getValue("type-id");
+
+	/**
+	 * Extract host, port, path from URL
+	 */
+
+	size_t findProtocol = url.find_first_of(":");
+	string protocol = url.substr(0,findProtocol);
+
+	string tmpUrl = url.substr(findProtocol + 3);
+	size_t findPort = tmpUrl.find_first_of(":");
+	string hostName = tmpUrl.substr(0, findPort);
+
+	size_t findPath = tmpUrl.find_first_of("/");
+	string port = tmpUrl.substr(findPort + 1 , findPath - findPort -1);
+	string path = tmpUrl.substr(findPath);
+
+	string hostAndPort(hostName + ":" + port);	
+
+	// Allocate connector struct
+	CONNECTOR_INFO* connInfo = new CONNECTOR_INFO;
+	// Set configuration felds
+	connInfo->hostAndPort = hostAndPort;
+	connInfo->path = path;
+	connInfo->timeout = timeout;
+	connInfo->typesId = typesId;
+	connInfo->producerToken = producerToken;
+	connInfo->formatNumber = formatNumber;
+	connInfo->formatInteger = formatInteger;
+
+	// Use compression ?
+	string compr = configCategory.getValue("compression");
+	if (compr == "True" || compr == "true" || compr == "TRUE")
+		connInfo->compression = true;
+	else
+		connInfo->compression = false;
+
+
+	// Log plugin configuration
+	Logger::getLogger()->info("%s plugin configured: URL=%s, "
+				  "producerToken=%s, OMF_types_id=%s, compression=%s",
+				  PLUGIN_NAME,
+				  url.c_str(),
+				  producerToken.c_str(),
+				  typesId.c_str(),
+				  connInfo->compression ? "True" : "False");
+
+	return (PLUGIN_HANDLE)connInfo;
+}
+
+/**
+ * Send Readings data to historian server
+ */
+uint32_t plugin_send(const PLUGIN_HANDLE handle,
+		     const vector<Reading *>& readings)
+{
+	CONNECTOR_INFO* connInfo = (CONNECTOR_INFO *)handle;
+        
+	/**
+	 * Allocate the HTTPS handler for "Hostname : port"
+	 * connect_timeout and request_timeout.
+	 * Default is no timeout at all
+	 */
+	connInfo->sender = new SimpleHttps(connInfo->hostAndPort,
+					   connInfo->timeout,
+					   connInfo->timeout);
+  
+	// Allocate the PI Server data protocol
+	connInfo->omf = new OMF(*connInfo->sender,
+				connInfo->path,
+				connInfo->typesId,
+				connInfo->producerToken);
+
+	// Set OMF FormatTypes  
+	connInfo->omf->setFormatType(OMF_TYPE_FLOAT,
+				     connInfo->formatNumber);
+	connInfo->omf->setFormatType(OMF_TYPE_INTEGER,
+				     connInfo->formatInteger);
+
+	// Send data
+	uint32_t ret = connInfo->omf->sendToServer(readings,
```

Comment:
Need to handle error return values?

Reply:
ret value is passed to the caller: it's not handled here.